### PR TITLE
Feature/refactor cql

### PR DESCRIPTION
**Pull request by laodouya**

### CODE REVIEW

File: cmd/cql/internal/create.go [hunk 1]
```diff
@@ -58,42 +69,89 @@ func init() {
...
 
+	configInit(cmd)
+
+	// create database
+	// parse instance requirement
+
 	txHash, dsn, err := client.Create(meta)
 	if err != nil {
-		ConsoleLog.WithError(err).Error("create database failed")
+		ConsoleLog.WithError(err).Error("Create database failed")
```
Comment (leventeliu) [inline cmd/cql/internal/create.go]: The logging message should not be capitalized according to our style guide.

File: cmd/cql/internal/create.go [hunk 2]
```diff
@@ -104,6 +162,26 @@ func runCreate(cmd *Command, args []string) {
 		}
 	}
 
-	ConsoleLog.Infof("the newly created database is: %#v", dsn)
-	fmt.Println(dsn)
+	fmt.Printf("\nThe newly created database is: %#v\n", dsn)
+	storeOneDSN(dsn)
+	fmt.Printf("The connecting string beginning with 'covenant://' could used as a dsn for `cql console`\n or any command, and website like https://web.covenantsql.io\n")
```
Comment (leventeliu) [inline cmd/cql/internal/create.go]: `covenant` -> `covenantsql`
`could used` -> `could be used`

Comment (leventeliu) [reply cmd/cql/internal/create.go]: And don't understand what it means 'and website like https://web.covenantsql.io'

File: cmd/cql/internal/generate.go [hunk 3]
```diff
@@ -178,12 +170,22 @@ func configGen() {
...
 		SetExitStatus(1)
 		return
 	}
 	fmt.Println("Generated config.")
 
+	fmt.Printf("\nConfig file:      %s\n", configFilePath)
+	fmt.Printf("Private key file: %s\n", privateKeyFile)
+	fmt.Printf("Public key's hex: %s\n", hex.EncodeToString(publicKey.Serialize()))
+
+	fmt.Printf("\nAny further command could costs PTC.\nYou can get some free PTC from: https://testnet.covenansql.io/wallet/%s\n", walletAddr)
```
Comment (leventeliu) [inline cmd/cql/internal/generate.go]: Consider to print multiple lines of message with raw string literal.

File: cmd/cql/internal/wallet.go [hunk 4]
```diff
@@ -59,36 +59,36 @@ func walletGen() {
...
+	} else {
+		fmt.Printf("\n\nwallet address: %s\n", walletGen())
 	}
 
-	if strings.ToLower(tokenName) == "all" {
+	var err error
+	if strings.ToLower(tokenName) == "" {
 		var stableCoinBalance, covenantCoinBalance uint64
 
 		if stableCoinBalance, err = client.GetTokenBalance(types.Particle); err != nil {
```
Comment (leventeliu) [inline cmd/cql/internal/wallet.go:83]: If the error is "no such balance", show the balance as 0, or tell the user "your account is not created in the TestNet, please apply tokens from our faucet first"

File: cmd/cql/internal/create.go [hunk 5]
```diff
@@ -18,38 +18,49 @@ package internal
...
+    node                   uint      // Target node number
+    space                  uint      // Minimum disk space requirement, 0 for none
+    memory                 uint      // Minimum memory requirement, 0 for none
+    load-avg-per-cpu       float     // Minimum idle CPU requirement, 0 for none
+    encrypt-key            string    // Encryption key for persistence data
+    eventual-consistency   bool      // Use eventual consistency to sync among miner nodes
+    consistency-level      float     // Consistency level, node*consistency_level is the node number to perform strong consistency
+    isolation-level        int       // Isolation level in a single node
+    gas-price              uint      // customized gas price
+    advance-payment        uint      // customized advance payment
```
Comment (leventeliu) [inline cmd/cql/internal/create.go]: Update usage message (the argument is not a 'meta info JSON string' now). Remove these fields introduction cause they will appear in the flags usage message.

File: cmd/cql/internal/create.go [hunk 6]
```diff
@@ -58,42 +69,89 @@ func init() {
...
+	cmd.Flag.Float64Var(&meta.LoadAvgPerCPU, "load-avg-per-cpu", 0, "Minimum idle CPU requirement, 0 for none")
+	cmd.Flag.StringVar(&meta.EncryptionKey, "encrypt-key", "", "Encryption key for persistence data")
+	cmd.Flag.BoolVar(&meta.UseEventualConsistency, "eventual-consistency", false, "Use eventual consistency to sync among miner nodes")
+	cmd.Flag.Float64Var(&meta.ConsistencyLevel, "consistency-level", 0, "Consistency level, node*consistency_level is the node number to perform strong consistency")
+	cmd.Flag.IntVar(&meta.IsolationLevel, "isolation-level", 0, "Isolation level in a single node")
+	cmd.Flag.Uint64Var(&meta.GasPrice, "gas-price", 0, "customized gas price")
+	cmd.Flag.Uint64Var(&meta.AdvancePayment, "advance-payment", 0, "customized advance payment")
+}
+
+func runCreate(cmd *Command, args []string) {
```
Comment (leventeliu) [inline cmd/cql/internal/create.go:97]: Also print help if extra arguments are provided, e.g., `len(args)>0`.

File: cmd/cql/internal/console.go [hunk 7]
```diff
@@ -322,6 +323,60 @@ func run(u *user.User) (err error) {
...
+					ConsoleLog.Error("invalid choice number")
+					SetExitStatus(1)
+					Exit()
+				}
+			}
+
+			//Set dsn
+			dsn = dsnArray[choice]
+		} else {
+			ConsoleLog.Error("nether local dsn storage exists nor a dsn string present")
```
Comment (leventeliu) [inline cmd/cql/internal/console.go]: You mean neither

File: cmd/cql/internal/create.go [hunk 8]
```diff
@@ -58,42 +55,95 @@ func init() {
...
+	if len(args) > 0 {
+		ConsoleLog.Error("create params should set by sepecific param name like -node")
 		SetExitStatus(1)
-		return
+		help = true
```
Comment (leventeliu) [inline cmd/cql/internal/create.go:101]: Maybe return here?

**Review by leventeliu — APPROVED** [review]
All good.

**Review by xq262144 — APPROVED** [review]

**Pull request merged**